src/UI/Component/ViewControl/Factory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\ViewControl;
+/**
+ * This is how the factory for UI elements looks.
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      Mode View Controls enable the switching between different aspects of some data. The different modes are mutually
+	 *      exclusive and can therefore not be activated at once.
+	 *   composition: >
+	 *      Button Groups are composed of Buttons switching between active and inactive states.
+	 *   effect: >
+	 *      Clicking on an inactive Button turns this button active and all other inactive. Clicking on an active button
+	 *      has no effect.
+	 *
+	 * rules:
+	 *   usage:
+	 *      1: At least one Button MUST always be active.
+	 *      2: Not more than one Button MUST be active.
+	 *   accessibility:
+	 *      1: The Button Group of the Mode View Control MUST cary the role-attribute "group".
+	 *      2: The Button Group Mode View Control MUST set an aria-label clearly describing it's functionality.
+	 *      3: The currently active Button must be labeled by setting aria-checked to "true".
+	 *
+	 * ---
+	 * @param    array $labelled_actions Set of labelled actions (string|string)[]. The label of the action is used as key, the action itself as value.
+	 *          The first of the actions will be activated by default.
+	 *
+	 * @return \ILIAS\UI\Component\ViewControl\Mode
+	 */
+	public function mode($labelled_actions);
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      Section View Controls enable the switching between different sections of some data. Examples are subsequent
+	 *      days/weeks/months in a calendar or entries in a blog.
```

Comment:
Just for curiosity: Would a pagination be some special kind of Section View Control?